### PR TITLE
Added AlertState

### PR DESCRIPTION
**Pull request by morkra**

### CODE REVIEW

File: src/SDK/SmartDetectorsSDK/Alert.cs [hunk 1]
```diff
@@ -44,5 +46,11 @@ protected Alert(string title, ResourceIdentifier resourceIdentifier)
         /// Gets the resource identifier that this Alert applies to.
         /// </summary>
         public ResourceIdentifier ResourceIdentifier { get; }
+
+        /// <summary>
+        /// Gets the alert's state. A Smart Detector can use this property to auto-resolve
+        /// alerts that were created previously by setting the property's value to <see cref="AlertState.Resolved"/>.
```
Comment (EranGon) [inline src/SDK/SmartDetectorsSDK/Alert.cs]: the detector cannot set this property, move this comment to the ctor

File: src/SDK/SmartDetectorsSDK/Alert.cs [hunk 2]
```diff
@@ -19,7 +19,8 @@ public abstract class Alert
         /// </summary>
         /// <param name="title">The Alert's title.</param>
         /// <param name="resourceIdentifier">The resource identifier that this Alert applies to.</param>
-        protected Alert(string title, ResourceIdentifier resourceIdentifier)
+        /// <param name="state">The alert's state</param>
+        protected Alert(string title, ResourceIdentifier resourceIdentifier, AlertState state)
```
Comment (EranGon) [inline src/SDK/SmartDetectorsSDK/Alert.cs]: >state [](start = 88, length = 5)

maybe add default value of Active?

File: test/analysis/SmartDetectorsAnalysisTests/AlertPresentationTests.cs [hunk 3]
```diff
@@ -31,6 +33,7 @@ public void TestInitialize()
         public void WhenProcessingAlertThenTheContractsAlertIsCreatedCorrectly()
         {
             ContractsAlert contractsAlert = this.CreateContractsAlert(new TestAlert());
+            Assert.AreEqual(ContractsAlertState.Active, contractsAlert.State);
```
Comment (EranGon) [inline test/analysis/SmartDetectorsAnalysisTests/AlertPresentationTests.cs]: >Active [](start = 48, length = 6)

add a test with resolved state

File: src/SDK/SmartDetectorsSDK/Alert.cs [hunk 4]
```diff
@@ -19,7 +19,9 @@ public abstract class Alert
         /// </summary>
         /// <param name="title">The Alert's title.</param>
         /// <param name="resourceIdentifier">The resource identifier that this Alert applies to.</param>
-        protected Alert(string title, ResourceIdentifier resourceIdentifier)
+        /// <param name="state">The alert's state. A Smart Detector can use this property to auto-resolve
+        /// alerts that were created previously by setting the property's value to <see cref="AlertState.Resolved"/>.</param>
```
Comment (EranGon) [inline src/SDK/SmartDetectorsSDK/Alert.cs]: this is not a property anymore, change to:
A Smart Detector can auto-resolve alerts that were created previously by passing <see cref="AlertState.Resolved"/> for this parameter

Comment (morkra) [reply src/SDK/SmartDetectorsSDK/Alert.cs]: Thanks

---
In reply to: [199709414](https://github.com/liadtal/Smart-Alerts/pull/176#discussion_r199709414) [](ancestors = 199709414)

**Pull request merged**